tools/build_pytorch_libs.py

```diff
@@ -275,7 +273,9 @@ def build_caffe2(version,
                  build_dir):
     my_env = create_build_env()
     build_test = not check_negative_env_flag('BUILD_TEST')
-    max_jobs = os.getenv('MAX_JOBS', None)
+    max_jobs = os.getenv('MAX_JOBS',
```

Comment:
This code refactor is changing the behavior for generators other than Ninja on Windows. However, I'm fine about this.

Reply:
I am suspicious here. I think the non windows builds do the fixup elsewhere and this is stealing a core of compilation from all developers.

Reply:
@zdevito  With respect to `max_jobs`, this change is equivalent to the version before this. All branches of the original piles of `if-else` clauses have changed `max_jobs` in this way, but written in inconsistent styles.

Reply:
I read the code again and I don't see how this is true. In the code before this patch if MAX_JOBS is not set in the environment, and the build is not windows, then max_jobs gets set to either not passed (for Ninja), or it is set to the core count. This is true for the majority of developers. When I look at my own build, it is not set.

Reply:
Yes, you are right, I realized I did neglect the case in which `MAX_JOBS` is not set, `IS_WINDOWS` is false and `USE_NINJA` is true. Which one of the following do you think would be the best, and I would modify the code accordingly:

- If `MAX_JOBS` is not set, then `max_jobs = number of CPUs`. Otherwise `max_jobs=MAX_JOBS`
- If `MAX_JOBS` is not set and `USE_NINJA` is false, then `max_jobs = number of CPUs`. Otherwise `max_jobs=MAX_JOBS`.

I'm personally leaning toward the first situation, because we would not need to specifically single out ninja and the purpose is nevertheless served.

Regarding the support of sccache (which would require one idle CPU if we support it according to the comment), I would say just ignore it in this script -- people who want to use sccache can manually set `MAX_JOBS` and we shouldn't let this single platform drag down the building speed of other platforms. What do you think?

Let me know and I'll change everything in one shot. Thanks!

Reply:
Yes, you are right, I realized I did neglect the case in which `MAX_JOBS` is not set, `IS_WINDOWS` is false and `USE_NINJA` is true. In this case, number of jobs is unspecified. But if `IS_WINDOWS` is changed to true, the number of jobs is specified, which does not make sense to me.

To resolve this, I would propose changing the behavior to either of the following. Which one of the following do you think would be the best, and I would modify the code accordingly:

- If `MAX_JOBS` is not set, then `max_jobs = number of CPUs`. Otherwise `max_jobs=MAX_JOBS`
- If `MAX_JOBS` is not set and `USE_NINJA` is false, then `max_jobs = number of CPUs`. Otherwise `max_jobs=MAX_JOBS`.

I'm personally leaning toward the first, because we would not need to specifically single out ninja and the purpose is nevertheless served.

Regarding the support of sccache (which would require one idle CPU if we support it according to the comment), I would say just ignore it in this script -- people who want to use sccache can manually set `MAX_JOBS` and we shouldn't let this single platform drag down the building speed of other platforms. What do you think?

Let me know and I'll change everything in one shot. Thanks!

Reply:
I'd prefer "If MAX_JOBS is not set, then max_jobs = number of CPUs. Otherwise max_jobs=MAX_JOBS". However, our CI uses clcache on windows which suffers from the same exact issues as sccache and will hang if we use all the cores. So if this change is made, it is also necessary to get MAX_JOBS set somewhere in our windows CI scripts to prevent that.

Reply:
I've made the change accordingly. It looks like in CI and builds `MAX_JOBS` is already properly handled, unless there are missing spots (e.g., [here](https://github.com/pytorch/pytorch/blob/990e63f587bd0bcaefb3d30a50594911f287d982/.jenkins/caffe2/build.sh#L91) and [here](https://github.com/pytorch/pytorch/blob/990e63f587bd0bcaefb3d30a50594911f287d982/.circleci/scripts/setup_ci_environment.sh#L58-L60)). It is still possible that I may have missed something. I think I'll just wait a bit to see whether all CI tests can pass...

Reply:
Those are not the windows build scripts though. Windows uses (or at least used to) use different scripts. Dead locks from not setting this flag are non-deterministic. Passing CI is not a guarantee that it is correct.

Reply:
OK, I have double checked again. I am aware of two Windows builds, one in `.jekins/pytorch` and one in `.jekins/caffe2`. I added a commit that fixes the one in `.jekins/pytorch`. Looks like Windows shares the same build script in `.jekins/caffe2` and [it's already handled](https://github.com/pytorch/pytorch/blob/990e63f587bd0bcaefb3d30a50594911f287d982/.jenkins/caffe2/build.sh#L91).